packages/api-reference/src/hooks/useNavigate.ts

```diff
@@ -0,0 +1,88 @@
+import { slug } from 'github-slugger'
+import { ref } from 'vue'
+
+import { type Heading, sleep } from '../helpers'
+import type { Tag, TransformedOperation } from '../types'
+
+type NavState = {
+  id?: string
+  label?: string
+}
+
+// Enables the interesection observer
+const canIntersect = ref(true)
+
+const navState = ref<NavState>({})
```

Comment:
I think we want this to be reactive not a ref? Depends on the setting logic but refs don't trigger on deep sets.

Reply:
In this case we only set the whole object so its the same? Style?